src/bin/stratis_dumpmetadata.rs

```diff
@@ -0,0 +1,57 @@
+use std::{
+    env,
+    fs::OpenOptions,
+    io::{Seek, SeekFrom},
+    process,
+    vec::Vec,
+};
+
+use serde_json::Value;
+
+use libstratis::engine::BDA;
+
+fn run(devpath: &str) -> Result<(), String> {
+    let mut devfile = OpenOptions::new()
+        .read(true)
+        .open(&devpath)
+        .map_err(|the_io_error| format!("Error opening device: {}", the_io_error))?;
+
+    let bda = BDA::load(&mut devfile)
+        .map_err(|bda_load_error| format!("BDA detected but error found: {}", bda_load_error))?
+        .ok_or_else(|| "No Stratis BDA metadata found".to_string())?;
+    println!("{:#?}", bda);
+
+    devfile
+        .seek(SeekFrom::Start(0))
+        .map_err(|seek_err| format!("Error during seek: {}", seek_err))?;
+
+    let loaded_state = bda
+        .load_state(&mut devfile)
+        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?;
+    println!("State JSON data:");
+
+    let state_json: Value = serde_json::from_slice(&loaded_state.unwrap())
```

Comment:
@mulkieran Should this `unwrap` be an `.expect()`?

Reply:
> @mulkieran Should this `unwrap` be an `.expect()`?

Whoops! Thanks for noticing.

@bgurney-rh This ```unwrap()``` is for the ```Option``` part of the return type of ```load_state()```. So you should handle it like you handled the ```Option``` part of the result type of ```load()```. And you should handle it around line 30, where you handle the ```Result``` part of the return type of ```load_state()```.